File: server/Models/Enums/Enums.cs
namespace server.Models.Enums;

public enum CalculationPeriod
{
    Monthly = 1,
    Quarterly = 2,
    Yearly = 3
}

public enum MetricSourceType
{
    Excel = 1,
    Manual = 2,
    ExternalApi = 3
}

public enum FieldDataType
{
    String = 1,
    Number = 2,
    Date = 3,
    Boolean = 4
}

public enum ImportStatus
{
    Pending = 1,
    Processing = 2,
    Success = 3,
    PartialSuccess = 4,
    Failed = 5
}

File: client/src/app/features/data-import/models/enums.ts
export type FieldDataType = 'String' | 'Number' | 'Date' | 'Boolean';

export type ImportStatus = 'Pending' | 'Processing' | 'Success' | 'PartialSuccess' | 'Failed';

export type MetricSourceType = 'Excel' | 'Manual' | 'ExternalApi';

export type CalculationPeriod = 'Monthly' | 'Quarterly' | 'Yearly';

export const IMPORT_STATUS_LABELS: Record<ImportStatus, string> = {
  Pending: 'ממתין',
  Processing: 'בעיבוד',
  Success: 'הושלם בהצלחה',
  PartialSuccess: 'הושלם עם שגיאות',
  Failed: 'נכשל',
};
